src/Utils/ResolvedType.php

```diff
@@ -69,19 +69,16 @@ public static function self(): ResolvedType
      *
      * @param string $typeName The resolved name of the type.
      */
-    public function __construct(Type $type, \Closure $fnToCode)
+    public function __construct(
+        /** @var Type *Readonly* The type of this resolved-type. */
+        public ReadOnly Type $type,
+        private \Closure $fnToCode,
```

Comment:
Wouldn't it be better to `use Closure` instead of `\Closure`?

Reply:
agreed. Done!